src/pages/api/preview.js

```diff
@@ -0,0 +1,36 @@
+import wp from "@/promisetracker/lib/wp";
+
+export default async function preview(req, res) {
+  const {
+    postType,
+    postId,
+    revisionId,
+    _wpnonce: nonce,
+    _thumbnail_id: thumbnailId,
+  } = req.query;
+  // Fetch WordPress to check if the provided `id` or `slug` exists
+  const wpApi = wp();
+  let post;
+  if (postType === "pages") {
+    post = await wpApi.revisions({ id: postId, nonce, revisionId, thumbnailId })
+      .page;
+  } else if (postType === "posts") {
+    post = await wpApi.revisions({ id: postId, nonce, revisionId, thumbnailId })
+      .post;
+  }
+  // If the post doesn't exist prevent preview mode from being enabled
+  if (!post) {
+    return res.status(404).redirect(`/not-found`);
+  }
+  res.setPreviewData({
+    query: { id: postId, nonce, revisionId, thumbnailId },
+  });
+
+  if (postType === "pages") {
+    res.redirect(`/about/${post.slug}`);
+  } else {
+    res.redirect(`/analysis/articles/${post.slug}`);
```

Comment:
Is it possible to also know and pass page slug from [`custom_preview_page_link`](https://github.com/CodeForAfrica/HURUmap-Dashboard/blob/e21644d3040e972902fe3be418f2fd9153b9329c/wp-content/themes/hurumap/functions.php#L224-L239) function? I'm thinking for those cases where we have more pages that need previewing than just `about` and `articles`.

Reply:
Or put differently if we have say `4` types of pages that need previewing, how would we decide which path to redirect to?

Reply:
Currently, the slug isn't always available, especially for draft posts, it's nonexistent. What I thought of, was converting the title to slugs, then we can get a switch statement for all the pages

Reply:
OK, thinking out load.

All **pages** will have slugs i.e. there is no way someone will want to preview a completely new (draft) page via our front ends. Put another way there will always be a known relationship between pages in the dashboard and pages in the front-end. So slug must always be available when previewing pages.

Now, what we are left with is **posts**. It is possible for someone to be previewing a draft article post (or myth busting post in outbreak.AFRICA speak).

So the question then becomes: Can we know the **parent** (article, myth busting, etc.) of the post so that we include that as slug? That should at least tell us where to redirect the preview to.

Reply:
@kilemensi  For this, we can't know because we assign the post to the parent page after creation.
I will suggest we start creating custom post types, instead of having parents.

Reply:
Cool,

1. @KhadijaMahanga , do you remember why we went with the parent / child relationship instead of custom types?
2. Should we decide to adopt the custom post type approach, can we easily retrofit existing posts to be of a custom type?

Reply:
Seen a few Plugins that can easily help us with that